File: june-20/Frontend/src/app/models/postmodel.ts
import { UserProfile } from "./userprofilemodel";
import { Category } from "./categorymodel";
import { Comment } from "./commentmodel";
import { Image } from "./imagemodel";
import { PostLike } from "./postlikemodel";
export interface Post {
  id?: string;  // optional since it’s generated server-side

  userEmail: string;
  title: string;
  slug: string;
  content: string;
  status?: string;         // default to "Published" if not provided
  isDeleted?: boolean;
  createdAt?:Date;
  views?:number;
  user?: UserProfile;             
  comments?: Comment[];
  categories?: Category[];
  images?: Image[];
  postLikes?:PostLike[];
}

File: june-20/Backend/models/postmodel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BlogPlatform.Models
{
    public class Post
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        [EmailAddress]

        public string UserEmail { get; set; }

        [Required]
        public string Title { get; set; }

        [Required]
        public string Slug { get; set; }

        [Required]
        public string Content { get; set; }

        public string Status { get; set; } = "Published"; // (Draft, Published, Deleted)
        public bool IsDeleted { get; set; } = false;

        public User User { get; set; }
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Category> Categories { get; set; } = new List<Category>();

        public ICollection<Image> Images { get; set; } = new List<Image>();
    }
}
